polkadot/xcm/xcm-builder/src/universal_exports.rs

```diff
@@ -710,3 +713,29 @@ mod tests {
 		}
 	}
 }
+
+/// An adapter for the implementation of `ExporterFor`, which attempts to find the
+/// `(bridge_location, payment)` for the requested `network` and `remote_location` and `xcm`
+/// in the provided `T` table containing various exporters.
+pub struct NetworkWithXcmExportTable<T, M>(core::marker::PhantomData<(T, M)>);
+impl<T: Get<Vec<NetworkExportTableItem>>, M: Contains<Xcm<()>>> ExporterFor
+	for NetworkWithXcmExportTable<T, M>
+{
+	fn exporter_for(
+		network: &NetworkId,
+		remote_location: &InteriorLocation,
+		xcm: &Xcm<()>,
+	) -> Option<(Location, Option<Asset>)> {
+		T::get()
+			.into_iter()
+			.find(|item| {
+				&item.remote_network == network &&
+					M::contains(xcm) &&
+					item.remote_location_filter
+						.as_ref()
+						.map(|filters| filters.iter().any(|filter| filter == remote_location))
+						.unwrap_or(true)
+			})
+			.map(|item| (item.bridge, item.payment))
```

Comment:
I would suggest a small optimization by reusing `NetworkExportTable`, so you don't need to duplicate code. Currently, your `M::contains(xcm)` (implemented by `xcm.clone().0.matcher()`) is called for every item in the list. However, the XCM instructions do not depend on the table, network, or remote location, so you are repeatedly matching the same XCM, which could be checked just once.

```suggestion
        // check `network` and `remote_location`
		let Some(bridge_with_fees) = NetworkExportTable::<T>::exporter_for(network, remote_location, xcm) else {
				return None,		
		};
		
		// check the XCM
		if M::contains(xcm) {
				Some(bridge_with_fees)
		} else {
				None
		}
```

And perhaps it makes sense to check the XCM beforehand and then `network, remote_location` with iterating `T` table.

Reply:
https://github.com/paritytech/polkadot-sdk/pull/6074/commits/606355c235f4d666334621f665f76b930fb2f35d